File: server/models/hashtagsonpost.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class hashtagsOnPost extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      this.belongsTo(models.posts);
      this.belongsTo(models.hashtags);
    }
  }
  hashtagsOnPost.init({
    postId: { type: DataTypes.INTEGER, allowNull: false },
    hashtagId: { type: DataTypes.INTEGER, allowNull: false },
    uuid: { type: DataTypes.UUID, allowNull: false, defaultValue: DataTypes.UUIDV4 }
  }, {
    sequelize,
    modelName: 'hashtagsOnPost',
  });
  return hashtagsOnPost;
};

File: lib/models/hashtag/hashtagsonpost.dart
class HashtagsOnPost {
  final int id;
  final int postId;
  final int hashtagId;
  final DateTime createdAt;
  DateTime updatedAt;

  HashtagsOnPost({
    required this.id,
    required this.postId,
    required this.hashtagId,
    required this.createdAt,
    required this.updatedAt,
  });
}
